distribution/osbuild-dnf-json.service

```diff
@@ -0,0 +1,17 @@
+[Unit]
+Description=OSBuild dnf-json service
+After=network.target
+Requires=osbuild-dnf-json.socket
+
+[Service]
+Type=simple
+PrivateTmp=true
+ExecStart=/usr/libexec/osbuild-composer/dnf-json
+Restart=always
+RestartSec=1s
+CacheDirectory=osbuild-dnf-json
+# systemd >= 240 sets this, but osbuild-worker runs on earlier versions
+Environment="CACHE_DIRECTORY=/var/cache/osbuild-dnf-json"
+
+[Install]
+WantedBy=osbuild-composer.service
```

Comment:
Should be wanted by the workers too though, no @teg ? Or we include `Wants=` or `Require=` in workers and composer rather than WantedBy here? not sure

Reply:
Good point! I didn't think of this. I don't think we need an install section at all, we should just require the socket from the service that depend on it. The service would be started on demand.

Reply:
Agreed with both of you, I suggest dropping the `Install` section and just directly using `Requires`, or `Wanted`. For now, I would just put `Requires` everywhere, it's imho the safest option.

Reply:
So should I make the worker require something ?

Reply:
The worker should require the `.socket`, and that should be all that's needed, I think.

Reply:
Hmm, when I add the dependency to the `.socket` systemd refuses to start the process because it's missing a dependency

Reply:
And also, if the osbuil-dnf-json.service does not require its socket, the activation stops working.

Reply:
> Hmm, when I add the dependency to the .socket systemd refuses to start the process because it's missing a dependency

Can you post diff of your changes and the message from log?

> And also, if the osbuil-dnf-json.service does not require its socket, the activation stops working.

I think that the dnf-json.service always has to require its socket. That shouldn't be changed.

Reply:
Both `osbuild-worker*.service` and `osbuild-dnf-json.service` should have `Requires=osbuild-dnf-json.socket`. I think that should be all that's needed.

Reply:
Do you think that `osbuild-remote-worker@.service` service too ? It is unclear because it does not have any requirements now.

Reply:
Yeah, anything that will connect to the socket needs to `Require` it.